### DISCUSSION TITLE
Fully decoupling a map from irl coordinates

### DISCUSSION THREAD

**Discussion by lunarvi** [visgl/deck.gl#7866] [Q&A] [answered]
Hey

A super basic version of what showcases my problem:

`
import React from 'react';
import DeckGL from '@deck.gl/react';
import {GeoJsonLayer} from '@deck.gl/layers';
import {COORDINATE_SYSTEM} from '@deck.gl/core';
import {createRoot} from "react-dom/client";
import provinces from "./provinces.json";

function App() {
    const INITIAL_VIEW_STATE = {
        latitude: 0,
        longitude: 0,
        zoom: 1,
        minZoom: 0,
        maxZoom: 12
    };
    function hexToRgb(hex) {
        hex = hex.split("#")[1];
        const bigint = parseInt(hex, 16);
        const r = (bigint >> 16) & 255;
        const g = (bigint >> 8) & 255;
        const b = bigint & 255;
        return [r, g, b];
    }

    const geojson = new GeoJsonLayer({
        id: 'geojson-layer',
        pickable: true,
        widthScale: 20,
        widthMinPixels: 2,
        data: provinces,
        stroked: true,
        filled: true,
        lineWidthScale: 2,
        lineWidthMinPixels: 2,
        getFillColor: f => {
            return hexToRgb(f.properties.hex);
        },
        getLineColor: [0, 0, 255, 128],
        getLineWidth: 1,
        coordinateSystem: COORDINATE_SYSTEM.IDENTITY,
    });
    return (
        <DeckGL
            initialViewState={INITIAL_VIEW_STATE}
            controller={true}
            layers={[geojson]}
        />
    );
```
}
export default App;
export function renderToDOM(container) {
    createRoot(container).render(<App/>);
}

````

The big tl;dr is: I'm "migrating" from Leaflet/D3, trying to use deckGL to render a videogame map, that is - a map with no real attachment to the real world. The map comprises of a GeoJSON FeatureCollection. 
While the above script seems to be enough to get deckGL to render each feature without any distortions, I seem to have ran into an insurmountable issue when it comes to panning/zooming boundaries. 
![image](https://user-images.githubusercontent.com/113045926/235490645-b7d94292-16c9-44f2-b71d-ea19a3afba9b.png)
With the above settings, the initial "camer" is set in the upper-left corner of the map + the size of the viewport(from what I can see)
The panning is almost entirely locked though. Movement on the Y axis is limited by what seems to be 50-100 pixels at most, movement on the X axis causes the entire view to reset past a certain point. (logging changes to viewState, it seems that deckGL maybe thinks you've reached the end of the map going by the longtitude, and tries to reset it? it goes from roughly +180 to -180 in any case

Some of the things I tried:
1. setting custom coordinate origin seems to be enough to move the initial camera position, but.. only on the Y axis, and only within the invisible boundaries of what deckGL thinks the longtitude of the map is
2. Removing all negative coordinates from the geojson(I thought that might be an issue for whatever reason), achieves the same problems, but on the other side of the map.
3. I managed to actually get the featurecollection to display somewhat properly via orthographicview, but that's not what I'm after.

If possible, I would really really appreciate some help with addressing this, as well as some general guidance in how I should approach developing my map. 

Thank you for reading and hope you have a wonderful day!

**Comment by lunarvi** [comment, marked as answer]
Of course I solve it the seconds after writing here :D 

The solution seems to be to use METER_OFFSETS, with the origin set to 0,0,0 rather than IDENTITY
Leaving this topic in case anyone else was running into this issue, and also in hopes that someone could share some more insights into "properly" building maps like this.

**Reply by Pessimistress** [reply]
The default view is MapView which uses Mercator projection. It also imposes WGS84 limitations to map interactions. If you are building a non-geographical map you should check out `OrthographicView` or `OrbitView`.

**Reply by mdanzige** [reply]
Sorry to bump an old thread.  I am also trying to figure out the best way to get MapView-style camera interactions with a non-geographic visualization.  I have a visualization that sits mostly in the Z = 0 plane, but is 3D (imagine an extruded 2D bar chart).  OrbitView with orthographic projection mostly gives me the behavior I'm looking for, except that as you pan and zoom with OrbitView, I find that the OrbitView camera target will quickly get "unprojected" to some point well below the Z = 0 plane, which creates some navigation and clipping issues. 

Naively, I think that if I could get OrbitView to lock its target Z position to 0, this would give me the behavior I want.  I believe this is something like what MapView does.  Would customizing OrbitView to override its usage of pixelsToWorld be the right approach here?  Or is there something else I'm missing?  I have tried adapting my visualization to MapView without much success, and I suspect I'm barking up the wrong tree there.
